math/dec.go

```diff
@@ -576,3 +577,68 @@ func MaxDec(x, y Dec) Dec {
 	}
 	return y
 }
+
+// MustAdd performs addition between two Dec values.
+// If an error occurs during the addition, it panics.
+func (x Dec) MustAdd(y Dec) Dec {
+	z, err := x.Add(y)
+	if err != nil {
+		panic("MustAdd failed: " + err.Error())
+	}
+	return z
+}
+
+// MustSub performs subtraction between two Dec values.
+// If an error occurs during the subtraction, it panics.
+func (x Dec) MustSub(y Dec) Dec {
+	z, err := x.Sub(y)
+	if err != nil {
+		panic("MustSub failed: " + err.Error())
+	}
+	return z
+}
+
+// MustMul performs multiplication between two Dec values.
+// If an error occurs during the multiplication, it panics.
+func (x Dec) MustMul(y Dec) Dec {
+	z, err := x.Mul(y)
+	if err != nil {
+		panic("MustMul failed: " + err.Error())
+	}
+	return z
+}
+
+// MulInt multiplies the decimal by an int64 and returns the result or an error.
+func (x Dec) MulInt(y int64) (Dec, error) {
+	var z Dec
+	if _, err := dec128Context.Mul(&z.dec, &x.dec, apd.New(y, 0)); err != nil {
+		return Dec{}, fmt.Errorf("MulInt failed: %w", err)
+	}
+	return z, nil
+}
+
+// MustMulInt multiplies the decimal by an int64 and panics if an error occurs.
+func (x Dec) MustMulInt(y int64) Dec {
+	z, err := x.MulInt(y)
+	if err != nil {
+		panic(err)
+	}
+	return z
+}
+
+// ToLegacyDec converts a Dec value to a LegacyDec value.
+// If the conversion fails, it returns an error.
+func (x Dec) ToLegacyDec() (LegacyDec, error) {
+	coeff := x.dec.Coeff
+	i := new(big.Int)
+
+	if _, ok := i.SetString(coeff.String(), 10); !ok {
```

Comment:
It is not enough to set the coeffizient only. 
For example:
```
		{"-1.23", NewDecWithExp(-123, -2), LegacyMustNewDecFromStr("-1.23")},
```
It would be good to add more scenarios to your test. Especially on the boundaries for regression.